File: liteadapter/src/main/java/me/yuu/liteadapter/loadmore/ILoadMoreFooter.java
package me.yuu.liteadapter.loadmore;

import androidx.annotation.IntDef;

import android.view.View;

/**
 * @author yu
 * @date 2018/1/12
 */
public interface ILoadMoreFooter {

    @IntDef({Status.LOADING, Status.COMPLETED, Status.NO_MORE, Status.ERROR})
    @interface Status {
        int LOADING = 0;
        int COMPLETED = 1;
        int NO_MORE = 2;
        int ERROR = 3;
    }

    @Status
    int getStatus();

    void setStatus(@Status int state);

    View getView();

}

File: liteadapter/src/main/java/me/yuu/liteadapter/loadmore/ILoadMoreFooter.kt
package me.yuu.liteadapter.loadmore

import android.view.View
import androidx.annotation.IntDef

/**
 * @author yu
 * @date 2018/1/12
 */
interface ILoadMoreFooter {

    @IntDef(Status.LOADING, Status.COMPLETED, Status.NO_MORE, Status.ERROR)
    annotation class Status {
        companion object {
            const val LOADING = 0
            const val COMPLETED = 1
            const val NO_MORE = 2
            const val ERROR = 3
        }
    }

    @get:Status
    var status: Int

    val view: View
}